readthedocs/api/v2/utils.py

```diff
@@ -209,7 +209,9 @@ def delete_versions_from_db(project, version_data):
             '(Sync Versions) Deleted Versions: project=%s, versions=[%s]',
             project.slug, ' '.join(deleted_versions),
         )
-        to_delete_qs.delete()
+        # Delete one by one to make sure the delete method is called.
+        for version in to_delete_qs.iterator():
+            version.delete()
```

Comment:
This seems like we're walking backwards on our version performance code. Won't this be slow when deleting a bunch of versions?

Reply:
yeah, but the slowness is mainly from github sending one event per branch deleted, thus creating a lot of sync versions tasks. This can be slow if the user switches the repo to one with fewer branches (like importing the stress git repo and then changing the repo to one with one branch).

Another alternatives I can think:

- Do the update in a scheduled task, like old builds from 100 versions each half hour or so.
- An scheduled task that deletes all orphan builds (builds without a version object), thus, only new builds will be kept, and old builds are going to be deleted when the version or project is deleted.